packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -441,18 +471,31 @@ export class LocalIntervalCollection<TInterval extends ISerializableInterval> {
         end: number,
         intervalType: MergeTree.IntervalType,
         props?: MergeTree.PropertySet) {
-        const interval = this.createInterval(start, end, intervalType);
+        const interval: TInterval = this.createInterval(start, end, intervalType);
         if (interval) {
             interval.addProperties(props);
             if (this.label && (this.label.length > 0)) {
```

Comment:
i'm not sure how safe just indexing into the properites object is below? will it always be defined? or should we clone the incoming props and set it all in addProperties
`{...props, [reservedIntervalIdKey]: uuid()}`